src/softmax.jl

```diff
@@ -26,32 +33,25 @@ julia> softmax([1, 2, 3])
 See also [`logsoftmax`](@ref).
 """
 function softmax(xs::AbstractArray; dims=1)
-    max_ = maximum(xs, dims=dims)
-    exp_ = exp.(xs .- max_)
-    exp_ ./ sum(exp_, dims=dims)
+    max_ = fmaximum(xs, dims=dims)
+    exp_ = fmap(exp, xs .- max_)
```

Comment:
Ah, yes, `@avx` broadcasting assumed that if `isone(stride(A,i))` is known at compile time, that it isn't being broadcast along that dimension.

Reply:
In about half an hour I'll release a new version of LoopVectorization that allows `LowDimArray{D,T,N}` to specify less than `N` dimensions in `D`. `LowDimArray{(false,)}(rand(1,5,6))` would then allow LoopVectorization's broadcasting to be aware that the first dimension is only `1`.

This could then be used as:
```julia
max_ = fmaximum(xs, dims=dims)
exp_ = if 1 ∈ dims
    @avx exp.(xs .- LowDimArrays{(false,)}(max_))
else dims isa Integer
    @avx exp.(xs .- max_)
end
```
(Although the issues about broadcasting being less robust about array types is still there.)